syntax = "proto3";

package greetings;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc TransferFile (FileRequest) returns (stream FileChunk) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

message FileRequest{
    string filename = 1;
}

message FileChunk{
    bytes content = 1;
}
